tools/build-machinery-status.py

```
@@ -0,0 +1,202 @@
+RUNNER_INDEX = re.compile(r"-\d+$")
+
+
+def netbox_get(api, token, path):
+    """GET an API path, following pagination, returning all results."""
+    results = []
+    url = f"{api}{path}"
+    while url:
+        req = urllib.request.Request(url, headers={
+            "Authorization": f"Token {token}",
+            "Accept": "application/json",
+        })
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)